src/net/sourceforge/kolmafia/FamiliarData.java

```diff
@@ -641,16 +641,14 @@ public void activate() {
     }
   }
 
-  private void changeWeight(int oldWeight, int newWeight) {
-    if (oldWeight == newWeight) {
-      return;
-    }
+  public void setWeight(final int weight) {
+    this.weight = weight;
     switch (this.id) {
       case FamiliarPool.GREY_GOOSE:
         if (this.active) {
-          if (oldWeight < 6 && newWeight >= 6) {
+          if (weight >= 6) {
             addGreyGooseSkills();
-          } else if (oldWeight >= 6 && newWeight < 6) {
+          } else if (weight < 6) {
```

Comment:
tiny comment: if weight is not >= 6, it is definitely less than 6 (right? there's no int NaN?), so you can do without the second "if".

Reply:
Fixed